netbox/dcim/views.py

```diff
@@ -3131,6 +3131,25 @@ def get_object(self, **kwargs):
 
         return obj
 
+    def get_extra_addanother_params(self, request):
+
+        params = {
+            'a_terminations_type': request.GET.get('a_terminations_type'),
+            'b_terminations_type': request.GET.get('b_terminations_type')
+        }
+
+        pk = resolve(request.GET.get('return_url')).kwargs.get('pk')
+        a_type = CABLE_TERMINATION_TYPES.get(request.GET.get('a_terminations_type'))
+
+        if hasattr(a_type, 'device'):
+            params.update({'termination_a_device': pk})
+        elif hasattr(a_type, 'power_panel'):
+            params.update({'termination_a_powerpanel': pk})
+        elif hasattr(a_type, 'circuit'):
+            params.update({'termination_a_circuit': pk})
```

Comment:
You could instead do:

```suggestion
for key in request.POST:
    if 'device' in key or 'power_panel' in key or 'circuit' in key:
        params.update({key: request.POST.get(key})
```

This lets you remove the resolve import

Reply:
Thank you for the suggestion. I have updated this PR as requested.